packages/strapi-helper-plugin/lib/src/components/LeftMenuSubList/index.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useState } from 'react';
+
+import { Collapse } from 'reactstrap';
+
+import Dropdown from './Dropdown';
+import LeftMenuLink from '../LeftMenuLink';
+
+const LeftMenuSubList = ({ name, links, isFiltered, isFirstItem }) => {
+  const [collapse, setCollapse] = useState(isFiltered || isFirstItem);
+  const [filtered, setFiltered] = useState(collapse);
+
+  const toggle = () => {
+    setCollapse(!collapse);
+  };
+
+  useEffect(() => {
```

Comment:
You are trying to synchronize your states in order to do this you can either use a reducer in the parent component so you can synchronise all your collapse or do the following.

(The following code doesn't keep the previous opened state but I don't think it's an issue)
```
const [collapse, setCollapse] = useState(false);
useEffect(() => {
  setCollapse(isSearching);
}, [isSearching]);

// Always open first one on new cycle
useEffect(() => {
  if (isFirstItem && !isSearching) {
    setCollapse(true);
  }
}, [isFirstItem, isSearching]);
```